tests/01_03_06-2wp_no_locking_cap.js

```diff
@@ -1,116 +1,79 @@
+ 
+  const fulfillRequirementsToRunAsSingleTestFile = async () => {
+    await rskUtils.activateFork(Runners.common.forks.wasabi100);
+  };
   
   before(async () => {
-    btcClient = bitcoin.getClient(
-      Runners.hosts.bitcoin.rpcHost,
-      Runners.hosts.bitcoin.rpcUser,
-      Runners.hosts.bitcoin.rpcPassword,
-      NETWORK
-    );
-    rskClient = rsk.getClient(Runners.hosts.federate.host);
-    rskClients = Runners.hosts.federates.map(federate => rsk.getClient(federate.host));
-    pegClient = pegUtils.using(btcClient, rskClient);
-    test = pegAssertions.with(btcClient, rskClient, pegClient);
-    whitelistTestLegacy = whitelistingAssertionsLegacy.with(btcClient, rskClient, pegClient);
-    rskTxHelpers = getRskTransactionHelpers();
-    btcTxHelper = getBtcClient();
+      rskTxHelpers = getRskTransactionHelpers();
+      rskTxHelper = rskTxHelpers[0];
+      btcTxHelper = getBtcClient()
 
-    // Grab the federation address
-    federationAddress = await rskClient.rsk.bridge.methods.getFederationAddress().call();
-    await btcClient.importAddress(federationAddress, 'federations');
-    
-    // Mine a few rsk blocks to prevent being at the beginning of the chain,
-    // which could trigger border cases we're not interested in
-    await sequentialPromise(10, () => rskUtils.mineAndSync(rskTxHelpers));
-    
-    // // Update the bridge to sync btc blockchains
-    await rskClient.fed.updateBridge();
-    await rskUtils.mineAndSync(rskTxHelpers);
+      if(process.env.RUNNING_SINGLE_TEST_FILE) {
+          await fulfillRequirementsToRunAsSingleTestFile();
+      }
+  })
 
-    return federationAddress;
-  });
 
   it('should transfer BTC to RBTC', async () => {
-    try {
-      var initialFedBalance = (await btcClient.getAddressBalance(federationAddress))[federationAddress] || 0;
-      expect(initialFedBalance).to.be.lessThan(bitcoin.btcToSatoshis(1000));
-      
-      const INITIAL_RSK_BALANCE = bitcoin.btcToSatoshis(1000) + bitcoin.btcToSatoshis(1);
-      const INITIAL_BTC_BALANCE = INITIAL_RSK_BALANCE + bitcoin.btcToSatoshis(1);
-      const TO_BRIDGE_GAS_PRICE = 1;
-      const WEIS_TO_RELEASE = '500000000000000000000';
+    const btcAddressInfo = await btcTxHelper.generateBtcAddress('legacy');
+    await whitelistingAssertions.assertAddLimitedLockWhitelistAddress(rskTxHelper, btcAddressInfo.address, 3 * MINIMUM_PEGIN_VALUE_IN_SATOSHI);
+    await rskUtils.mineAndSync(rskTxHelpers);
+    
+    const recipientRskAddressInfo = getDerivedRSKAddressInformation(btcAddressInfo.privateKey, btcTxHelper.btcConfig.network);
+    await rskTxHelper.importAccount(recipientRskAddressInfo.privateKey);
+    const unlocked = await rskTxHelper.unlockAccount(recipientRskAddressInfo.address);
+    expect(unlocked, 'Account was not unlocked').to.be.true;
 
-      var addresses = await pegClient.generateNewAddress('test');
-      expect(addresses.inRSK).to.be.true;
+    const latestActiveForkName = getLatestActiveForkName();
```

Comment:
Remember this is an async function, you need to call it with `await`, otherwise it will return a promise. Right now this test is working because the `getBridge` function uses the default abi (which is the latest) when it is not passed a valid fork name. And that is not what we want.

Reply:
hmm, i removed it because sonarcloud was complaining about it... 
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=5&id=rsksmart_rootstock-integration-tests&open=AYpM6CRVPbiqigSZLqn8

so how can i pass the sonarcloud checks with the await here?

Reply:
updated JSDoc of `getLatestActiveForkName()`